probe/pkg/runtime/runtime.go

```
@@ -0,0 +1,71 @@
+package runtime
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"github.com/stackrox/acs-fleet-manager/probe/config"
+	"github.com/stackrox/acs-fleet-manager/probe/pkg/probe"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var backoff = wait.Backoff{
+	Duration: 1 * time.Second,
+	Factor:   1.5,
+	Jitter:   0.1,
+	Steps:    15,
+	Cap:      10 * time.Minute,
+}
+
+// Runtime performs a probe run against fleet manager.
+type Runtime struct {
+	Config *config.Config
+}
+
+// New creates a new runtime.
+func New() (*Runtime, error) {
+	config, err := config.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load configuration")
+	}
+
+	return &Runtime{
+		Config: config,
+	}, nil
+}
+
+// RunLoop a continuous loop of probe runs.
+func (r *Runtime) RunLoop(ctx context.Context) error {
+	ticker := time.NewTicker(r.Config.RuntimeRunWaitPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.Wrap(ctx.Err(), "probe context invalid")
+		case <-ticker.C:
+			err := r.RunSingle(ctx)
+			glog.Warning(err)
+		}
+	}
+}
+
+// RunSingle executes a single probe run.
+func (r *Runtime) RunSingle(ctx context.Context) error {
+	ctxTimeout, cancel := context.WithTimeout(ctx, r.Config.RuntimeRunTimeout)
+	defer cancel()
+	defer r.CleanUp()
```

I saw @misberner's comment suggesting to change this later on - specifically making the call to `r.CleanUp()` async and expose e.g. a concurrency.Signal indicating whether cleanup has been finalized. So I can assume it's a TODO for later on?

Irrespective of the changes to be made, it's probably still worthwhile to use `defer utils.Should(r.CleanUp())` instead of ignoring the error from `r.CleanUp()` here.

---

Yeah I'll do this in the next iteration, want to get this one done to avoid moving target.